android/quest/src/test/java/org/smartregister/fhircore/quest/util/extensions/ConfigExtensionsTest.kt

```diff
@@ -729,9 +750,108 @@ class ConfigExtensionsTest : RobolectricTest() {
     defaultRepository.create(addResourceTags = true, binaryImage)
     loadRemoteImagesBitmaps(
       listOf(columnProperties.children[0]),
-      computedValuesMap = emptyMap(),
       registerRepository = registerRepository,
+      computedValuesMap = emptyMap(),
+      configurationRegistry.decodedImageMap,
     )
     assertNotNull(imageProperties.imageConfig?.decodedBitmap)
   }
+
+  @Test
+  fun testImageMapNotUpdatedWhenReferenceIsNull() = runTest {
+    val cardViewProperties = profileConfiguration.views[0] as CardViewProperties
+    val listViewProperties = cardViewProperties.content[0] as ListProperties
+    val columnProperties = listViewProperties.registerCard.views[0] as ColumnProperties
+    val rowProperties =
+      (columnProperties.children[0] as RowProperties).copy(
+        children =
+          listOf(
+            ImageProperties(
+              imageConfig =
+                ImageConfig(
+                  type = ICON_TYPE_REMOTE,
+                  reference = null,
+                ),
+            ),
+          ),
+      )
+    val emptyComputedValuesMap = mutableMapOf<String, String>()
+    val decodedImageMap = mutableMapOf<String, Bitmap>()
+
+    val logMessages = mutableListOf<String>()
+    Timber.plant(
+      object : Timber.Tree() {
+        override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
+          logMessages.add(message)
+        }
+      },
+    )
+
+    loadRemoteImagesBitmaps(
+      listOf(rowProperties),
+      registerRepository = registerRepository,
+      computedValuesMap = emptyComputedValuesMap,
+      decodedImageMap = decodedImageMap,
+    )
+
+    assertTrue(logMessages.isEmpty())
+    assertTrue(decodedImageMap.isEmpty())
+  }
+
+  @Test
+  fun testExceptionCaughtOnDecodingBitmap() = runTest {
```

Comment:
You can pass the expected parameter to the Test annotation to verify that an exception is thrown. e.g. `@Test(expected = NullPointerException.class)`

Reply:
@ellykits 
do we have to throw the exception for this particular test?

Reply:
That depends on the expected exceptions, null pointer exception was one of them according to your previous implementation.